pkg/chunk/storage/caching_index_client.go

```diff
@@ -233,14 +237,15 @@ func (s *cachingIndexClient) cacheFetch(ctx context.Context, keys []string) (bat
 		var readBatch ReadBatch
 
 		if err := proto.Unmarshal(bufs[j], &readBatch); err != nil {
-			level.Warn(util.Logger).Log("msg", "error unmarshalling index entry from cache", "err", err)
+			level.Warn(log).Log("msg", "error unmarshalling index entry from cache", "err", err)
 			cacheCorruptErrs.Inc()
 			continue
 		}
 
 		// Make sure the hash(key) is not a collision in the cache by looking at the
 		// key in the value.
 		if key != readBatch.Key || (readBatch.Expiry != 0 && time.Now().After(time.Unix(0, readBatch.Expiry))) {
+			level.Warn(log).Log("msg", "collision/expiration in entry cache", "key", key, "readBatch.Key", readBatch.Key, "expiry", time.Unix(0, readBatch.Expiry))
```

Comment:
Why is this a warning?
Can we have the message be more actionable for the operator who reads it?

Reply:
I've split these two reasons into separate ifs & logs, and made it debug level.  WDYT?